GH-4008: Improve ReplyingKafkaTemplateTests with modular validation

snowykte0426 opened the pull request:
## Summary
  This PR addresses issue #4008 by refactoring `ReplyingKafkaTemplateTests`
   to improve test maintainability and readability through better
  separation of concerns.

  - Refactored listener methods to return `Message<?>` objects instead of
  handling validation internally
  - Extracted header validation logic into separate utility methods
  (`validateHeaders`, `validateSuccessfulResponse`)
  - Simplified test methods by moving validation concerns outside of
  listener logic

  ## Changes Made
  - **Listener Methods**: Simplified
  `handleCustomReplyHeaderNoReplyPartition` and
  `handleCustomReplyHeaderDefaultPartitionHeader` to focus only on business
   logic
  - **Validation Utilities**: Added `validateHeaders()` and
  `validateSuccessfulResponse()` methods for external header verification
  - **Test Methods**: Updated `testCustomReplyTopicHeaderIsNotDuplicated`
  and `testCustomReplyHeadersAreNotDuplicated` to use the new validation
  approach

  Fixes #4008

artembilan reviewed (changes requested):
Thank you for refactoring this hard-to-read logic!

snowykte0426 commented on spring-kafka/src/test/java/org/springframework/kafka/requestreply/ReplyingKafkaTemplateTests.java:
```diff
@@ -932,12 +936,15 @@ void testMessageIterableReturn() throws Exception {
+
+	private void validateSuccessfulResponse(Message<?> message, String expectedPayload) {
+		assertThat(message.getPayload()).isEqualTo(expectedPayload);
+		assertThat(message.getHeaders()).containsKey("originalPayload");
```
> If we do assert this header in the `validateHeaders()`, then no need to duplicate it here. And in the end we would not need this extra method then. To be honest, it is hard for me to understand the test logic and maintain it afterwards when there are some custom methods. I really prefer to stick with just `assertThat()` in place. And now imaging the situation that one of those tests would be modified eventually and this custom method (read `validateHeaders()`) would not work any more. What I mean, that tests are essentially independent units of work, so trying to avoid duplication does not pay off with a maintenance time.
> 
> But that's my opinion.

Thank you for the feedback, @artembilan — you're absolutely right about keeping test assertions direct and clear.

I've updated the implementation in commit 5ca3d81 to remove the custom validation methods (`validateHeaders()` and `validateSuccessfulResponse()`) and replaced them with inline `assertThat()` statements. After reviewing the existing codebase and seeing similar tests use just a few repeated assertions, I realized the inline approach is much more intuitive.

This change makes each test’s intent immediately obvious without needing to jump to separate methods. Thanks again for pointing me in the right direction!

snowykte0426 commented on spring-kafka/src/test/java/org/springframework/kafka/requestreply/ReplyingKafkaTemplateTests.java:
```diff
@@ -392,7 +392,10 @@ public void testCustomReplyTopicHeaderIsNotDuplicated() throws Exception {
+			
 			assertThat(resultingMessage.getPayload()).isEqualTo("OK");
+			assertThat(resultingMessage.getHeaders()).containsKey("originalPayload");
+			assertThat(resultingMessage.getHeaders().get("originalPayload")).isEqualTo("expected_message");
```
@artembilan Thank you for the suggestion! I’ve implemented your feedback in commit e8baed9 by using AssertJ’s containsEntry() method. This replaces the separate assertions with the more concise approach you recommended.

artembilan reviewed (changes requested):
Thank you for the update!
It is really very close to merge.
However I’ve noticed the problem:

> Signed-off-by: snowykte0426 <snowykte0426@naver.com>

this one has to come with your official name to satisfy all the legal requirements.
Please, revise your local Git client config.
Squash and push a commit with with right sign-off.

snowykte0426 commented:
> Thank you for the update! It is really very close to merge. However I’ve noticed the problem:
> 
> > Signed-off-by: snowykte0426 [snowykte0426@naver.com](mailto:snowykte0426@naver.com)
> 
> this one has to come with your official name to satisfy all the legal requirements. Please, revise your local Git client config. Squash and push a commit with with right sign-off.

Thank you for the review feedback. I've squashed the commits and updated
  the Signed-off-by with my official name as requested (commit b965e86).
